RecomputePathing errors at end of map load

Delegate OnRecomputePathing - Standard for message type RecomputePathing failed with exception 
Object reference not set to an instance of an object
  at BattleTech.PathNodeGrid.ResetPathGrid (UnityEngine.Vector3 origin, System.Single beginAngle, BattleTech.PathingCapabilitiesDef pcd, System.Single maxDistance, BattleTech.MoveType moveType) [0x00239] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at BattleTech.Pathing.ResetPathGridIfTouching (System.Collections.Generic.List`1[T] Rectangles, UnityEngine.Vector3 origin, System.Single beginAngle, BattleTech.AbstractActor actor) [0x0006b] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at BattleTech.AbstractActor.OnRecomputePathing (MessageCenterMessage msg) [0x00064] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at MessageCenter.SendMessagesForType (MessageCenterMessageType messageType, MessageCenterMessage message) [0x00037] in <d9d0f3cf980a449a8897715cd4df053c>:0 
2019-11-09T22:20:22 FYLS [ERROR] CRITICAL ERROR, PLEASE REPORT:
Delegate OnRecomputePathing - Standard for message type RecomputePathing failed with exception 
Object reference not set to an instance of an object
  at BattleTech.PathNodeGrid.ResetPathGrid (UnityEngine.Vector3 origin, System.Single beginAngle, BattleTech.PathingCapabilitiesDef pcd, System.Single maxDistance, BattleTech.MoveType moveType) [0x00239] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at BattleTech.Pathing.ResetPathGridIfTouching (System.Collections.Generic.List`1[T] Rectangles, UnityEngine.Vector3 origin, System.Single beginAngle, BattleTech.AbstractActor actor) [0x0006b] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at BattleTech.AbstractActor.OnRecomputePathing (MessageCenterMessage msg) [0x00064] in <d9d0f3cf980a449a8897715cd4df053c>:0 
  at MessageCenter.SendMessagesForType (MessageCenterMessageType messageType, MessageCenterMessage message) [0x00037] in <d9d0f3cf980a449a8897715cd4df053c>:0 

keep getting this stacktrace currently, will keep digging what may cause it

--- a/src/Core/PathFinderManager.cs
+++ b/src/Core/PathFinderManager.cs
@@ -311,6 +311,8 @@ namespace MissionControl {
     }
 
     public void Reset() {
+      UnsubscribePathfinders();
+
       if (pathFinderMech.GameRep != null) {
         GameObject pathFinderGo = pathFinderMech.GameRep.gameObject;
         GameObject blipUnknownGo = pathFinderMech.GameRep.BlipObjectUnknown.gameObject;
@@ -336,6 +338,11 @@ namespace MissionControl {
       Reset();
       pathFinderMech = null;
       pathFinderVehicle = null;
+    }
+
+    private void UnsubscribePathfinders() {
+      AccessTools.Method(typeof(Mech), "SubscribeMessages").Invoke(pathFinderMech, new object[] { false });
+      AccessTools.Method(typeof(AbstractActor), "SubscribeMessages").Invoke(pathFinderVehicle, new object[] { false });
     }
   }
 }

